plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/api.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { UrlReader, resolveSafeChildPath } from '@backstage/backend-common';
+import { promises as fs } from 'fs-extra';
+import { examples } from './api.examples';
+import { createTemplateAction } from '@backstage/plugin-scaffolder-node';
+
+import { CatalogApi } from '@backstage/catalog-client';
+import path from 'path';
+import { AuthService } from '@backstage/backend-plugin-api';
+
+/**
+ * Gets the definition of an API and places it in the workspace, or optionally
+ * in a subdirectory specified by the 'targetPath' input option.
+ * @public
+ */
+export function createFetchApiAction(options: {
+  reader: UrlReader;
+  catalog: CatalogApi;
+  auth?: AuthService;
+}) {
+  const { reader, catalog, auth } = options;
+
+  return createTemplateAction<{
+    apiEntityRef: string;
+    targetPath: string;
+  }>({
+    id: 'fetch:api',
```

Comment:
We already have an action called `catalog:fetch` I think that we should rename this to `catalog:fetch:api` or something similar to keep the namespacing sane.